app/models/billing_expense.rb

```diff
@@ -19,4 +21,25 @@ def billing_hours
   def billed_hours
     Hour.where(billing_expense: id)
   end
+
+  def auto_set_amount
+    self.amount = compute_amount
+  end
```

Comment:
Any reason to split this across two methods?

Reply:
To avoid writing self.amount 4 times. Basically... no real reason

Reply:
You can also do

```ruby
self.amount =
  if
    ...
  elsif
    ...
  end
```